### ISSUE TITLE
Privacy concerns about third-party scrapers

### ISSUE THREAD

**Issue by alexweissman** [gitterHQ/gitter#1314]
I know that [privacy has been an issue](https://github.com/gitterHQ/gitter/issues/332) with regards to gitter before, but this is something that hopefully the gitter team at least recognizes is a problem: third-party scrapers.

I'm not going to link to them because I don't want to give them any more of an SEO boost, but a site calling itself `gitterforum.com` has apparently been scraping public Gitter chat rooms and hosting the contents on their Google-indexed site.  I appreciate that Gitter itself has the option to opt-out of having chat archives indexed by search engines, but of course third-party sites are going to do whatever they want.  This particular third-party site apparently doesn't care about Gitter users' privacy, despite that they use a ["whois guard" for their own domain](https://whois.icann.org/en/lookup?name=gitterforum.com).

I realize that this is largely out of Gitter's direct control, and that the very nature and design of Gitter makes this a thorny problem.  It is certainly convenient for legitimate users to have public access to a chat room history without needing to sign in, but it comes at a very real cost of privacy.  The nature of chat is very informal.  I do my best to avoid saying things that I don't want effectively etched in stone forever, but the truth is that the experience of chat "feels" like a conversation between friends, and our users might say things - while using their real names - that they don't necessarily want to pop up when someone Googles them.

I don't necessarily mind that what we say in chat is publicly accessible, or even that a public log of that is preserved forever.  I try to keep our dialogues as professional as possible.  The problem is that when combined with search engine indexing, it takes an intimate (if not private) setting and pushes it to the forefront of the public domain.

I don't know if there is an easy solution to this.  I would be happy if there were an ability to limit the visibility of my group's chat contents (via both the frontend interface and API) to users who have actually joined the chat.  This would at least make it more difficult for third-party sites who are just indiscriminately downloading all of Gitter's archives.

**Comment by MadLittleMods** [comment]
Hey @alexweissman,

Messages in public rooms are public and can be accessed at any time. Anyone can grab the messages in a public room using the API, https://developer.gitter.im/docs/welcome

On Gitter's side, you can turn off search engine indexing on the archives page(`https://gitter.im/org/repo/archives/all`) but we can't control how others display messages. Also, search engines can just as easily ignore the setting.

![](http://i.imgur.com/fxXgOPD.png)

If you are concerned about someone finding your posts via archive or search engine, it is probably best not to use a public room.

**Comment by alexweissman** [comment]
Yes, I saw in Gitter's [Terms of Service](https://gitter.zendesk.com/hc/en-us/articles/200944641-Terms-of-Service):

> You acknowledge and agree that public messages and other user generated content:
    can be viewed by all users and are indexed by search engines;
    are accessible for an unlimited period of time; and
    are subject to the Creative Commons licence: Attribution + Noncommercial + ShareAlike (BY-NC-SA).

It is a tricky area because we want to make rooms publicly accessible, but we don't want third parties mirroring our content and submitting to search engine indices.  A clause in the TOS forbidding bots from mirroring content might at least discourage parties like `gitterforum.com`.

At any rate, because of this we have decided to abandon gitter for our discussion needs, and set up a private chat server of our own where we can have better control of access and forbid logging/scraping.
